cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -236,11 +274,12 @@ private void handleEC2Exception(List<CloudInstance> vms, AmazonEC2Exception e) t
             return fillCloudVmInstanceStatuses(ac, cloudInstancesWithInstanceId, region, result);
         } catch (AmazonEC2Exception e) {
             handleEC2Exception(vms, e);
+            // The next line is pointless. Artifact of the way handleEC2Exception behaves.
+            return Collections.emptyList();
         } catch (SdkClientException e) {
             LOGGER.warn("Failed to send request to AWS: ", e);
             throw e;
         }
-        return Collections.emptyList();
```

Comment:
Not in this review, but we could do something like the following.

```
diff --git a/cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java b/cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java
index 849292bb71..bee055082d 100644
--- a/cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java
+++ b/cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java
@@ -111,7 +111,7 @@ public class AwsInstanceConnector implements InstanceConnector {
                 consumer.accept(amazonEC2Client, instances);
             }
         } catch (AmazonEC2Exception e) {
-            handleEC2Exception(vms, e);
+            throw handleEC2Exception(vms, e);
         } catch (SdkClientException e) {
             LOGGER.warn(exceptionText, e);
             throw e;
@@ -196,7 +196,7 @@ public class AwsInstanceConnector implements InstanceConnector {
         }
     }
 
-    private void handleEC2Exception(List<CloudInstance> vms, AmazonEC2Exception e) throws AmazonEC2Exception {
+    private AmazonEC2Exception handleEC2Exception(List<CloudInstance> vms, AmazonEC2Exception e) throws AmazonEC2Exception {
         LOGGER.debug("Exception received from AWS: ", e);
         if (e.getErrorCode().equalsIgnoreCase(INSTANCE_NOT_FOUND_ERROR_CODE)) {
             Pattern pattern = Pattern.compile("i-[a-z0-9]*");
@@ -207,7 +207,7 @@ public class AwsInstanceConnector implements InstanceConnector {
                 vms.removeIf(vm -> doesNotExistInstanceId.equals(vm.getInstanceId()));
             }
         }
-        throw e;
+        return e;
     }
 
     @Retryable(
@@ -235,12 +235,11 @@ public class AwsInstanceConnector implements InstanceConnector {
             LOGGER.debug("Result from AWS: {}", result);
             return fillCloudVmInstanceStatuses(ac, cloudInstancesWithInstanceId, region, result);
         } catch (AmazonEC2Exception e) {
-            handleEC2Exception(vms, e);
+            throw handleEC2Exception(vms, e);
         } catch (SdkClientException e) {
             LOGGER.warn("Failed to send request to AWS: ", e);
             throw e;
         }
-        return Collections.emptyList();
     }
```

Reply:
Yup. Should probably not have made this change in this PR. Was bothering me when reading the code.